packages/xo-web/src/xo-app/backup/restore/index.js

```diff
@@ -184,9 +143,58 @@ export default class Restore extends Component {
       backupDataByVm[vmId].backups = orderBy(backups, 'timestamp', 'desc')
     })
 
-    this.setState({ backupDataByVm })
+    return backupDataByVm
+  }
+
+  _refreshBackupListOnRemote = async (remote, jobs) => {
+    const remoteId = remote.id
+    const backupsByRemote = await listVmBackups([remoteId])
+    const { backupDataByVm } = this.state
+    const remoteBackupDataByVm = {}
+    forEach(backupsByRemote[remoteId], (vmBackups, vmId) => {
+      if (vmBackups.length === 0) {
+        return
+      }
+
+      const backupData = backupDataByVm[vmId]
+      remoteBackupDataByVm[vmId] =
+        backupData === undefined ? { backups: [] } : backupData
```

Comment:
You need to clone `backupData` to not share the same address with `remoteBackupDataByVm[vmId]` and to avoid to mutate a property of the state. See https://github.com/vatesfr/xen-orchestra/blob/342871deafeb2eb392e0ab1f50881246a4d108a6/packages/xo-web/src/xo-app/backup/restore/index.js#L163